cordova-lib/src/cordova/plugin.js

```diff
@@ -295,14 +307,87 @@ module.exports = function plugin(command, targets, opts) {
             }).then(function() {
                 return hooksRunner.fire('after_plugin_search');
             });
+        case 'save':
+            // save the versions/folders/git-urls of currently installed plugins into config.xml
+            return save(projectRoot, opts);
         default:
             return list(projectRoot, hooksRunner);
     }
 };
 
+function save(projectRoot, opts){
+    var xml = cordova_util.projectConfig(projectRoot);
+    var cfg = new ConfigParser(xml);
+
+    // First, remove all pre-existing plugins from config.xml
+    cfg.getPluginIdList().forEach(function(plugin){
+        cfg.removePlugin(plugin);
+    });
+
+    // It might be the case that fetch.json file is not yet existent.
+    // for example: when we have never ran the command 'cordova plugin add foo' on the project
+    // in that case, there's nothing to do except bubble up the error
+    function onFileNotFoundException(err){
+        return Q.reject(err.message);
+    }
+
+    // Then, save top-level plugins and their sources
+    return Q().then(function(){
```

Comment:
Since that there is no async code in this function, IMHO it'll be better to avoid wrapping it into a promise and just return fullfilled/rejected promise.